assignment2/src/com/assignment2/SaveUserDetails.java

```diff
@@ -0,0 +1,198 @@
+		User.choice2 = s.nextInt();
+	}
+	public void run() {
+		String fileName = "file.txt";
+		Scanner sc = new Scanner(System.in);
+		SaveUserDetails obj = new SaveUserDetails();
+		ArrayList<User> users = new ArrayList<User>();
+		boolean flag = true;
+		//Reading data from disk
+		try
+        {
+            FileInputStream fis = new FileInputStream(fileName);
+            ObjectInputStream ois = new ObjectInputStream(fis);
+ 
+            users = (ArrayList) ois.readObject();
+ 
+            ois.close();
+            fis.close();
+        }
+        catch (IOException ioe)
+        {
+            ioe.printStackTrace();
+            return;
+        }
+        catch (ClassNotFoundException c)
+        {
+            System.out.println("Class not found");
+            c.printStackTrace();
+            return;
+        }
+		while(flag == true) {
+			System.out.println("Select the operation you want to perform-:");
+			System.out.println("1. Add User");
+			System.out.println("2. Display Users");
+			System.out.println("3. Delete User");
+			System.out.println("4. Save Users to the disk");
+			System.out.println("5. Exit");
+			
+			System.out.println("Please Enter the number of your choice:");
+			int option = sc.nextInt();
+			switch(option) {
+			case 1: 
+					User user = null;
+					user = obj.userInput();
+					int number = user.getRollNo();
+					boolean flag1 = true;
+					for(User user1:users) {
+						if(user1.getRollNo() == number) {
+							flag1 = false;
+						}
+					}
+					if(flag1==true)users.add(user);
+					else System.out.println("Roll Number Already Exists! please try again");
+					Collections.sort(users);
+				break;
+			case 2:
+				obj.askUserToDisplay();
+				Collections.sort(users);
+				System.out.println("--------------------------------------------------------------------------------------------------------------------------");
+				//System.out.println("Name                   Age                      Address                  Roll No.                Courses                  ");
+				System.out.printf("%20s %10s %25s %10s %15s", "Name", "Age", "Address", "Roll No.", "Courses");
+				System.out.println();
+				System.out.println("---------------------------------------------------------------------------------------------------------------------------");
+				
+					for(User user1:users) {
+						System.out.printf("%20s %10s %25s %10s %15s",user1.getName(), user1.getAge(), user1.getAddress(), user1.getRollNo(), user1.courses);
+						System.out.println();
+					}
+				break;
+			case 3:
+				System.out.println("Enter the Roll Number of the student to delete:");
+				int number1 = sc.nextInt();
+				for (User u : users) {
+					if(u.getRollNo() == number1) {
+						users.remove(users.indexOf(u));
+						System.out.println("User Details Deleted Successfully!!");
+						break;
+					}
+				}
+				break;
+			case 4:
+				try
+		        {
+		            FileOutputStream fos = new FileOutputStream(fileName);
```

Comment:
Since the read/write operations are invoked in many places, could this be written as a different Operations class that works on Users type of object.